explainer.md

```diff
@@ -362,21 +363,147 @@ merchants that accept credit cards.
+credentials for an identified user, either legimitately or illicitly) to obtain
+a payment assertion. They then send that assertion to the Relying Party's
+**login** end-point, and hope that the Relying Party does not follow the
+[WebAuthn requirements on assertion verification][webauthn-verification].
+
+SPC does make sure that payment assertions differ from login assertions, in the
+following ways:
+
+1. The CollectedClientData `type` member is "payment.get", not "webauthn.get"
+1. The CollectedClientData `origin` member will be the calling origin, which
+     would **not** be the Relying Party in such an attack.
+1. The CollectedClientData will also have an additional `payment` member for
+     an SPC-generated assertion.
+
+In addition to the above, a Relying Party should also record what type of
+interaction (i.e. login or payment) a given `challenge` is generated for and
+ensure that the use of the assertion matches the expected interaction type.
+
+**Payment attack**
+
+In this attack, a malicious third-party tries to use SPC (again with either
+legimitately or illictly obtained credentials) to initiate an unauthorized
+payment. The attack is infeasible:
+
+* When the attacker initiates SPC, the user will be shown an interface by the
+    User Agent that clearly states the transaction details (including the payee
+    and amount). The user is very likely to reject this interface as invalid.
+* If the user does agree to the transaction, and completes the subsequent
+    WebAuthn authentication ceremony, the attacker now has a signed SPC
+    assertion for the Relying Party.
+* If the Relying Party is not expecting a transaction, it will reject the
+    assertion.
+* If the Relying Party is expecting a transaction, it will detect an
+    unfamiliar `challenge` and reject the assertion.
+
+The general concept is that a Secure Payment Confirmation assertion is
+essentially useless unless it is part of an ongoing online transaction.
+
+### Merchant-supplied authentication data
+
+In this attack, a malicious Merchant attempts to trick the Customer or Account
+Provider into agreeing to a different transaction than they are actually
+entering into, by passing in spoof values to the SPC API call. The Merchant is
+able to influence the:
+
+* Transaction amount and currency
+* Payment instrument name and icon
+* Payee origin
+
+For example, the Merchant could tell the Account Provider (in the backend) that
+it is initiating a purchase of $100, but then pass $1 to the SPC API (and thus
+show the Customer a $1 transaction to verify).
+
+This form of attack already exists on the web today (where an Account Provider
+largely has to trust that the Merchant is showing the correct details), and SPC
+actually makes it harder to perform this attack. After receiving an assertion,
+the Account Provider can (and should) verify the payment data in the assertion
+to ensure it aligns with the transaction details provided to the Account
+Provider by the Merchant.
```

Comment:
The text as written does not explain how the Account Provider previously received information about the transaction from the Merchant. My understanding is that this is implementation dependent, e.g., when integrating SPC with 3DS, transaction info is provided early to the ACS, then validated later by the ACS. Perhaps something like this:

This form of attack already exists on the web today (where an Account Provider largely has to trust that the Merchant is showing the correct details).  We anticipate that SPC will provide better protection against this attack when used according to this common pattern:

* The Merchant provides some "pre-authentication" transaction information (including merchant name and amount) to the Account Provider over backend rails.
* After receiving an assertion, the Account Provider can (and should) compare the pre-authentication details with the assertion details to ensure they align.